File: bildunya_frontend/lib/data/models/send_message_request.dart
/// [com.bildunya.dto.SendMessageRequest]
class SendMessageRequest {
  const SendMessageRequest({
    required this.conversationId,
    required this.content,
  });

  final int conversationId;
  final String content;

  Map<String, dynamic> toJson() => {
        'conversationId': conversationId,
        'content': content,
      };
}

File: BilDunya/src/main/java/com/bildunya/dto/SendMessageRequest.java
package com.bildunya.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SendMessageRequest {

    @NotNull(message = "Conversation id is required")
    private Long conversationId;

    @NotBlank(message = "Content is required")
    @Size(max = 2000, message = "Content must be at most 2000 characters")
    private String content;
}

